react/private/FieldMessage/FieldMessage.js

```diff
@@ -45,13 +45,19 @@ export default class FieldMessage extends Component {
 
     if (message) {
       const { critical, positive, secondary, ...restMessageProps } = this.props.messageProps;
+      const isCritical = (valid === false && !secondary) || critical;
+      const isPositive = (valid === true && !secondary) || positive;
+      const isSecondary = typeof valid === 'undefined' || secondary;
 
       return (
         <Text
           {...restMessageProps}
-          critical={(valid === false && !secondary) || critical}
-          positive={(valid === true && !secondary) || positive}
-          secondary={typeof valid === 'undefined' || secondary}>
+          critical={isCritical}
+          positive={isPositive}
+          secondary={isSecondary}
+          {...(isCritical && !restMessageProps.role) && { role: 'alert' }}
```

Comment:
Yes that's correct.  Just had a chat to Matt about it and he also finds this a little confusing to read so I am going to change it to a ternary and return an empty object in the case the condition on the left is false.